### PR TITLE
Expose sets and compounds

### PR DESCRIPTION
**Pull request by pjaspers**
This adds in all the backend related controllers to expose sets and compounds to be consumed by the `hesabu-manager`, it's more JSON API related controllers, without breaking any current functionality. 

I've added a few helpers in the models to be able to expose more information.

### CODE REVIEW

File: app/models/package.rb [hunk 1]
```diff
@@ -158,6 +158,33 @@ def target_entity_groups
...
+  # this package. This is a expensive call, that will probably be not
+  # needed anymore when the simulation ouput gets splitted into many
+  # parts.
+  #
+  # TODO: Refactor away the expensiveness
+  def simulation_org_unit
+    org_unit_group = main_entity_groups.first&.organisation_unit_group_ext_ref
+    return nil unless org_unit_group
+
+    Pyramid.from(project).org_units_in_group(org_unit_group).first
```
Comment (mestachs) [inline app/models/package.rb]: can take the "AutoComplete" way select in the snaphots directly ?

Comment (pjaspers) [reply app/models/package.rb]: That would be a lot faster, I thought I couldn't get it from the snapshot. Will investigate.

Comment (mestachs) [reply app/models/package.rb]:
```
select * from ( 
       select 
         (element->'table'->>'id')::text as id , 
             (element->'table'->>'display_name')::text as display_name, 
            (element->'table'->>'code')::text as code, element ->'table' as eleme 
         from ( 
              select jsonb_array_elements(content) as element 
              from dhis2_snapshots 
              where 
              id=( 
                 SELECT max(id) FROM dhis2_snapshots 
                 WHERE kind= 'organisation_units' AND project_anchor_id = 94) 
              ) 
          as elements 
       ) as dataelements 
       where EXISTS ( 
     SELECT TRUE  
     FROM jsonb_array_elements(eleme->'organisation_unit_groups') x  
     WHERE x->>'id' IN ('hrwKYSFVbdq')) limit 1               
```

Comment (pjaspers) [reply app/models/package.rb]: I'm going to work this out in a new PR

**Comment by pjaspers** [comment]
(rebased against latest dev, now waiting for build, if green will merge)

**Pull request merged**